src/esw/cameras.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python3
+
+from mrover.msg import CameraCmd
+from typing import List, Dict, Tuple
+
+import threading
+
+import rospy
+from mrover.srv import (
+    ChangeCameras,
+    ChangeCamerasRequest,
+    ChangeCamerasResponse,
+)
+
+import jetson.utils
+
+
+class VideoDevices:
+
+    device: int
+    resolution: List[str]
+    video_source: jetson.utils.videoSource
+    # Screens video source is displayed on
+    output_by_endpoint: Dict[str, jetson.utils.videoOutput]
+
+    def __init__(self, device: int):
+        self.resolution = []
+        self.device = device
+        self.video_source = None
+        self.output_by_endpoint = {}
+
+    def remove_endpoint(self, endpoint: str) -> None:
+        """
+        Removes screen endpoint from video source list.
+        If endpoint dictionary is empty, it also deletes video source.
+        :param endpoint: the endpoint (e.g. 10.0.0.7:5000)
+        """
+        assert endpoint in self.output_by_endpoint.keys()
+        del self.output_by_endpoint[endpoint]
+        if len(self.output_by_endpoint) == 0:
+            self.video_source = None
```

Comment:
Do we need to call video_source.close()? Or does that happen in the object's destructor?

Reply:
Don't need to.

Reply:
wait actually i havent tested in a while

Reply:
don't need to.